File: Leetcode/WaterBottles.rs
// https://leetcode.com/problems/water-bottles

struct Solution {}

impl Solution {
    pub fn num_water_bottles(num_bottles: i32, num_exchange: i32) -> i32 {
        let mut ans = num_bottles;
        let mut ex = num_bottles;

        while ex >= num_exchange {
            let add = ex / num_exchange;
            ans += add;
            ex = add + ex % num_exchange;
        }

        ans
    }
}

File: Leetcode/WaterBottles.c
// https://leetcode.com/problems/water-bottles

int numWaterBottles(int numBottles, int numExchange) {
    int ans = 0;
    int empty = 0;

    while (numBottles) {
        ans += numBottles;
        int all = empty + numBottles;
        numBottles = all / numExchange;
        empty = all % numExchange;
    }

    return ans;
}